java/tools/src/java/org/apache/orc/tools/convert/JsonReader.java

```diff
@@ -333,15 +364,20 @@ public JsonReader(Reader reader,
                     FSDataInputStream underlying,
                     long size,
                     TypeDescription schema,
-                    String timestampFormat) throws IOException {
-    this(new JsonStreamParser(reader), underlying, size, schema, timestampFormat);
+                    String timestampFormat,
+                    String unionTag,
+                    String unionValue) throws IOException {
+    this(new JsonStreamParser(reader), underlying, size, schema, timestampFormat, unionTag,
+         unionValue);
   }
 
   public JsonReader(Iterator<JsonElement> parser,
                     FSDataInputStream underlying,
                     long size,
                     TypeDescription schema,
-                    String timestampFormat) throws IOException {
+                    String timestampFormat,
+                    String unionTag,
+                    String unionValue) throws IOException {
```

Comment:
@rizaon. Should the original constructor be retained so that the code that already exists can be seamlessly compatible looks better. I suggest adding a new constructor to handle the case of specifying unionTag/unionValue

Reply:
Done